tests/compute/half-opaque-convert.slang

```diff
@@ -0,0 +1,28 @@
+//DISABLE_TEST(compute):COMPARE_COMPUTE:-dx12 -compute -use-dxil -profile cs_6_2 -render-features half -shaderobj
+//DISABLE_TEST(compute):COMPARE_COMPUTE:-vk -compute -profile cs_6_2 -render-features half -shaderobj
+//TEST(compute):COMPARE_COMPUTE:-cuda -compute -render-features half -shaderobj
+
+//DISABLE_TEST(compute):SIMPLE:-target ptx -stage compute -entry computeMain
+
+// The following test shows if half can be processed as an opaque type. This means
+// it can be copied, and converted to and from float types, but nothing else. 
+// No maths, no swizzling.
```

Comment:
It should be allowed to load a `half` from a `StructuredBuffer<half>`, right?

Reply:
In principal yes - although I need to check for sure as DX (and to a lesser degree GLSL) are somewhat vague around the half/float16 type. On CUDA half should be fine - but half3 is an issue in that the definition in prelude is 8 bytes, but perhaps it should be 6 but be slower.